### DISCUSSION TITLE
USBケーブルを右から左に変更したい

### DISCUSSION THREAD

**Discussion by katsusuke** [Yowkees/keyball#197] [Q&A] [answered]
Keyball44  rev1 を利用しております。
素晴らしいキーボードをありがとうございます。
一点困っていることがあり質問をさせてください。

USBケーブルを、右側のキーボードに接続して利用しているのですが、私の使っているノートPCには左側にしかUSBポートが無いので、左側に指して利用したいです。
ところが、USBケーブルを左側のキーボードに指して利用すると、キーボードでasdf を入力すると ;lkj が入力されてしまいます。

https://github.com/Yowkees/keyball/blob/main/keyball44/doc/rev1/buildguide_jp.md#%EF%BC%91%EF%BC%91keyball%E3%81%AE%E4%BD%BF%E3%81%84%E6%96%B9

> USBケーブルは左右どちらに差しても使用できます。ただしキーマップはUSBケーブルを差した側のProMicroへ書き込まれるため、初めて差し替えた場合にはキーマップを見直してください

上記の記載があるのですが、

> 初めて差し替えた場合にはキーマップを見直してください

これは、左手のキーボード (asdf....) を 右手のキーボード (;lkj ...) に交換したキーマップを自分で作成する必要があるということでしょうか?
それともハードや設定書き込み方法? 等を工夫すれば設定ができるということでしょうか?

自力で探すことができず質問してしまいました。
よろしくおねがいします

**Comment by d-hamasato** [comment, marked as answer]
Keyball39 rev2、QMKでファームウェアをビルドという私の環境での参考になってしまうのですが、
> USBケーブルを左側のキーボードに指して利用すると、キーボードでasdf を入力すると ;lkj が入力

このような現象は私の環境では起きていません。
左右どちらにUSB接続した場合でもasdf を入力するとasdf と入力されています。

すでにお試しになられた後かもしれないのですが、以下の2点を確認してみると改善するかもしれません。
1. 左右に同じファームウェアを再度書き込んでみる
    - 片方がテスト用のファームウェアのままということはないでしょうか？
2. [キーボード左右判定用のジャンパ](https://github.com/Yowkees/keyball/blob/main/keyball44/doc/rev1/buildguide_jp.md#3-5%E3%82%B8%E3%83%A3%E3%83%B3%E3%83%91%E3%81%AE%E3%81%AF%E3%82%93%E3%81%A0%E4%BB%98%E3%81%91)が正しく半田付けされているか確認する
    - 左右の判定がない左右対称の分割キーボードの場合、USBを左右差し替えると入力されるキーが反転するということが起きます。（そのようなキーボードでUSB接続したいサイドを変更する場合、QMKなどでmasterとなるサイドを変更します）
    - 上記の症状と同じように思えたので左右判定に関わるジャンパの半田付けが怪しいと感じました。

私はKeyball39ユーザーなのであくまで参考程度に、気が向いたらご確認いただければと思います。

**Comment by katsusuke** [comment]
ありがとうございます！
無事に解決することができました。
左右判定用(LEFT) のジャンパー自体は半田づけできていたのですが、その横のチップダイオードが浮いていました。

> 左右どちらにUSB接続した場合でもasdf を入力するとasdf と入力されています。

この一文が解決の決め手になりました。

下記のスクショの部分を再度はんだ付けし直すことで、左右を正常に判定できるようになりました。
![IMG_8049](https://user-images.githubusercontent.com/624736/226086057-f4de1929-c94a-42be-a6bd-153f44d8dfe0.jpg)

快適なキーボードライフを楽しませていただきます。
本当にありがとうございました！

本件はクローズさせていただきます。
